File: Elitesportweb/models/Notification.js
import mongoose from 'mongoose';

const NotificationSchema = new mongoose.Schema({
  title: { type: String, required: true },
  message: { type: String, required: true },
  classId: { type: mongoose.Schema.Types.ObjectId, ref: 'Class' },
  className: { type: String, required: true },
  type: { type: String, enum: ['general', 'important', 'urgent'], default: 'general' },
  scheduledDate: { type: Date },
  scheduledTime: { type: String },
  memberId: { type: mongoose.Schema.Types.ObjectId, ref: 'Member' },
  isSent: { type: Boolean, default: false },
  createdAt: { type: Date, default: Date.now },
  isActive: { type: Boolean, default: true }
});

export default mongoose.models.Notification || mongoose.model('Notification', NotificationSchema);

File: Elitesportweb/elite_sports_flutter/lib/models/notification.dart
class NotificationModel {
  final String id;
  final String title;
  final String message;
  final String className;
  final DateTime createdAt;

  NotificationModel({
    required this.id,
    required this.title,
    required this.message,
    required this.className,
    required this.createdAt,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      message: json['message'] ?? '',
      className: json['className'] ?? '',
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}